### DISCUSSION TITLE
Achievements Flags

### DISCUSSION THREAD

**Discussion by ScottFromDerby** [RetroAchievements/RAWeb#746] [Ideas]
`[start of meleu's edition]`
I'm hijacking this topic to make it address "Achievement Flags" in general (not only missables). [Click here to go directly to the comment](https://github.com/RetroAchievements/RAWeb/issues/225#issuecomment-424311111)
`[end of meleu's edition]`

Occasionally an achievement can be awarded for a task unrelated to completing the game, and there might be one opportunity to get it; it is otherwise "miss-able". Being able to somehow flag a game as Missable would quickly warn players who want to master a set.

See https://www.trueachievements.com/flags.aspx

**Comment by meleu** [comment]
Seems to be related to #221 

Initially I tended to be resistant to this type of idea for the same reasons @rzumer [commented on #221](https://github.com/RetroAchievements/RAWeb/issues/221#issuecomment-422561057) (specially the part he mentions that it's a burden for the achievement creator, who IMO already has a heavy load of work to produce a neat set).

But after reading that trueachievements link I saw something I liked:

> (...) we rely on you, the community, to assist us. If you've played a game and have any flagging suggestions or updates please feel free to submit them using the 'Submit Game Information' button (...)

So yeah, I think we could have this. Seems to be something to be addressed on @luchaos spreadsheet about new roles.

**Comment by ScottFromDerby** [comment]
Something I like about the TA implementation is that flags are completely explicit, very black and white. Either it's missable or it's not, or it's difficulty-specific, or it's not

**Comment by meleu** [comment]
Yeah. and using flags instead of "categories" seems to be easier to handle. Cause, for example, a cheevo can be "missable" AND "tough challenge" at the same time.

**Comment by kdecks** [comment]
Great way to handle it.

**Comment by meleu** [comment]
[I'm hijacking this topic to make it address "Achievement Flags" in general (not only missables).]

It would be cool to have achievement flags system (sometimes known as "tags") to be used on achievements and make it clear that they are "missable", "challenging", "easter eggs", etc...

The flags would be added via web interface and we should have some way to allow user contributions.

Users would be able to submit tags to an achievement and some dev/moderator/admin/whatever can approve it or not.

Important requisites:
- optional. This feature is cool, but it's just a cosmetic, right?
- rely on community contributions.
- do not consider "flag management" as the achievement set creator's responsibility

Inspiration: https://www.trueachievements.com/flags.aspx

Related issue: #221

**Comment by ScottFromDerby** [comment]
Sometimes achievements have videos, which can be found on the achievement page. I'd like to see a small icon on the game page, next to each achievement to indicate that a video is available.

[Rough mockup](https://i.imgur.com/lTO26f1.png)
